src/category.rs

```diff
@@ -1,57 +1,244 @@
-use quick_xml::{XmlReader, XmlWriter, Element, Event};
-use quick_xml::error::Error as XmlError;
+// This file is part of rss.
+//
+// Copyright © 2015-2017 The rust-syndication Developers
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the MIT License and/or Apache 2.0 License.
 
+use error::Error;
 use fromxml::FromXml;
+use quick_xml::{Element, Event, XmlReader, XmlWriter};
+use quick_xml::error::Error as XmlError;
 use toxml::ToXml;
-use error::Error;
+use url::Url;
 
 /// A representation of the `<category>` element.
 #[derive(Debug, Default, Clone, PartialEq)]
-pub struct Category {
+pub struct Category
+{
     /// The name of the category.
-    pub name: String,
+    name: String,
     /// The domain for the category.
-    pub domain: Option<String>,
+    domain: Option<String>,
 }
 
-impl FromXml for Category {
+impl Category
+{
+    /// Get the category that exists under `Category`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::{CategoryBuilder, Category};
+    ///
+    /// let category = "podcast";
+    ///
+    /// let category_obj = CategoryBuilder::new()
+    ///     .name(category)
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// assert_eq!(category.to_owned(), category_obj.name());
+    /// ```
+    pub fn name(&self) -> String
```

Comment:
how do you feel about making these getters returned borrowed instead of owned structures? e.g. `pub fn name(&self) -> &str`